sequencesender/txbuilder/banana_types.go

```diff
@@ -147,3 +148,34 @@ func (b *BananaSequence) LastVirtualBatchNumber() uint64 {
 func (b *BananaSequence) SetLastVirtualBatchNumber(batchNumber uint64) {
 	b.SequenceBanana.LastVirtualBatchNumber = batchNumber
 }
+
+func CalculateMaxL1InfoTreeIndexInsideL2Data(l2data []byte) (uint32, error) {
+	batchRawV2, err := state.DecodeBatchV2(l2data)
+	if err != nil {
+		return 0, fmt.Errorf("CalculateMaxL1InfoTreeIndexInsideL2Data: error decoding batchL2Data, err:%w", err)
+	}
+	if batchRawV2 == nil {
+		return 0, fmt.Errorf("CalculateMaxL1InfoTreeIndexInsideL2Data: batchRawV2 is nil")
+	}
+	maxIndex := uint32(0)
+	for i := range batchRawV2.Blocks {
+		if batchRawV2.Blocks[i].IndexL1InfoTree > maxIndex {
+			maxIndex = batchRawV2.Blocks[i].IndexL1InfoTree
+		}
```

Comment:
Minor optimization: use the block loop variable instead of indexing the slice twice. WDYT?

```suggestion
	for _, block := range batchRawV2.Blocks {
		if block.IndexL1InfoTree > maxIndex {
			maxIndex = block.IndexL1InfoTree
```